exporter/otlp/lib/opentelemetry/exporter/otlp/exporter.rb

```diff
@@ -341,6 +341,9 @@ def as_otlp_span_kind(kind)
 
         def as_otlp_key_value(key, value)
           Opentelemetry::Proto::Common::V1::KeyValue.new(key: key, value: as_otlp_any_value(value))
+        rescue Encoding::UndefinedConversionError => e
+          OpenTelemetry.handle_error(exception: e, message: "encoding error for key #{key} and value #{value}")
```

Comment:
I think this should be enough information to track down the source of these errors in a production environment.